cmd/notary/tuf.go

```diff
@@ -269,56 +265,49 @@ func verify(cmd *cobra.Command, args []string) {
 	return
 }
 
-func passphraseRetriever() (string, error) {
-	fmt.Println("Please provide a passphrase for this root key: ")
-	var passphrase string
-	_, err := fmt.Scanln(&passphrase)
-	if err != nil {
-		return "", err
-	}
-	if len(passphrase) < 8 {
-		fmt.Println("Please use a password manager to generate and store a good random passphrase.")
-		return "", errors.New("Passphrase too short")
-	}
-	return passphrase, nil
-}
+func passphraseRetriever(keyName string, createNew bool, numAttempts int) (string, bool, error) {
+	fmt.Printf("Retrieving passphrase for key %s: ", keyName)
 
-func getPassphrase(confirm bool) ([]byte, error) {
-	if pass := os.Getenv("NOTARY_ROOT_PASSPHRASE"); pass != "" {
-		return []byte(pass), nil
+	if (numAttempts > 3 && !createNew) {
+		return "", true, errors.New("Too many attempts")
 	}
 
 	state, err := term.SaveState(0)
 	if err != nil {
-		return nil, err
+		return "", false, err
 	}
 	term.DisableEcho(0, state)
 	defer term.RestoreTerminal(0, state)
 
 	stdin := bufio.NewReader(os.Stdin)
 
-	fmt.Printf("Enter root key passphrase: ")
+	fmt.Printf("Enter %s key passphrase: ", keyName)
```

Comment:
Let's customize the prompt to make it clear whether we're setting a new password or entering a password to unlock an existing key.